examples/chart/teleport-cluster/charts/teleport-operator/Chart.yaml

```diff
@@ -0,0 +1,10 @@
+.version: &version "11.0.0-dev"
+
+name: teleport-operator
```

Comment:
> the operator chart by itself does nothing and won't work if a user just hem installs

I think the subchart name is wrong but the pattern might still be relevant. Most projects are bundling the CRDs directly in the charts as you're suggesting. I think this might be a common antipattern.

Users adopting the operator pattern are deploying multiple operators, with cross-operator dependencies that are even sometimes cyclic. For example:
- The teleport-cluster chart deploys a `ServiceMonitor` CR, managed by the prometheus-operator to be monitored
- The prometheus-operator deploys a `TeleportApplcation` CR to register its Prometheus UIs in Teleport if the `TeleportApplication` CRD exists, else it does nothing.

In this example the operator deployment order matters and we can end up with different final outcomes. This is in contradiction with Kubernete's declarative approach and can lead to degraded features until all operators are restarted at least once.

Such situation can be solved by deploying first all CRDs, then all operators. Such pattern heavily simplifies new cluster deployment. Elastic is doing so with their elastic-cloud operator.

Maybe my past experience with operators is too and most of our users don't care, I'm totally ok with moving the CRDs back to the `teleport-cluster` chart. What's your opinion ?